src/Config/Haskell.hs

```diff
@@ -55,7 +56,7 @@ readSetting s (FunBind _ [Match _ (Ident _ (getSeverity -> Just severity)) pats
         let (a,b) = readSide $ childrenBi bind in
         let unit = GHC.noLoc $ GHC.ExplicitTuple GHC.noExt [] GHC.Boxed in
         [SettingMatchExp $
-         HintRule severity (head $ snoc names defaultHintName) s (fromParen lhs) (fromParen rhs) a b
+         HintRule severity (NEE.head $ names NEE.|: defaultHintName) s (fromParen lhs) (fromParen rhs) a b
```

Comment:
Would adding `headDef` not be a clearer semantics here? We can define a local `headDef` in a Util module.

Reply:
How about putting that in `Data.List.Extra`?

Reply:
Hmm... Maybe. But where do you stop? Maybe head/tail is exactly where you stop, because that's the one step constructor selectors. I'm not sure.

Reply:
I opened ndmitchell/extra#50. I'll revisit these once it closes.